colossus/src/main/scala/colossus/core/IOSystem.scala

```diff
@@ -15,42 +16,70 @@ import scala.concurrent.{ExecutionContext, Future}
+    * A Config object will be created via {{{ConfigFactory.load()}}}
+    * @param sys
+    * @return
+    */
+  def apply()(implicit sys : ActorSystem) : IOSystem = {
+    apply(ConfigRoot)
   }
 
-  def apply(name: String = "iosystem", numWorkers: Option[Int] = None)(implicit sys: ActorSystem = ActorSystem(name)): IOSystem = {
-    val workers = numWorkers.getOrElse (Runtime.getRuntime.availableProcessors())
-    val metrics = MetricSystem(MetricAddress.Root / name)
-    apply(name, workers, metrics)
+  /**
+    * Create a new IOSystem using the supplied configPath.  This configPath will be overlaid on top of the default "colossus.io-system"
+    * config path.
+    * A Config object will be created via {{{ConfigFactory.load()}}}
+    *
+    * @param configPath  The path to the configuration.
+    * @param sys
+    * @return
+    */
+  def apply(configPath : String)(implicit sys : ActorSystem) : IOSystem = {
+    apply(configPath, ConfigFactory.load())
   }
 
-}
+  /**
+    * Create a new IOSystem by loading its config from the specified configPath.  This configPath will be overlaid on top of the default "colossus.io-system" config path.
+    * @param configPath The path to the configuration
+    * @param config The Config source to query
+    * @param sys
+    * @return
+    */
+  def apply(configPath : String, config : Config)(implicit sys : ActorSystem) : IOSystem = {
+    val ioConfig = config.getConfig(configPath).withFallback(config.getConfig(ConfigRoot))
+    val name = ioConfig.getString("name")
+    val workerCount : Option[Int] = if(config.hasPath("num-workers")){Some(config.getInt("num-workers"))} else{ None }
+    val ms = MetricSystem(s"$configPath.metrics", config)
+    apply(name, workerCount, ms)
+  }
 
-/**
- * Configuration used to specify an IOSystem's parameters.
- * @param name The name of the IOSystem
- * @param numWorkers The amount of Worker actors to spawn.  Must be greater than 0.
- */
-case class IOSystemConfig(name: String, numWorkers: Int){
-  require (numWorkers >= 0) //FIX - we currently need to allow 0 workers for tests
+  /**
+    * Create a new IOSystem
+    * @param name Name of this IOSystem.  This will also be used as its MetricAddres.
+    * @param workerCount Number of workers to create
+    * @param metrics The MetricSystem used to report metrics
+    * @param system
+    * @return
+    */
+  def apply(name : String, workerCount : Option[Int], metrics : MetricSystem)(implicit system : ActorSystem) : IOSystem = {
+    val numWorkers = workerCount.getOrElse(Runtime.getRuntime.availableProcessors())
+
+    val workerManager = system.actorOf(Props(classOf[WorkerManager], WorkerManagerConfig(numWorkers, metrics)), s"iosystem-${actorFriendlyName(name)}-manager")
+    val sys = IOSystem(workerManager, name, numWorkers, metrics, system)
+    workerManager ! WorkerManager.Initialize(sys)
+    sys
+  }
+  private def actorFriendlyName(name : String) = {
```

Comment:
I think it's ok to leave it for now since it defaults to just "/".  As long as we don't force users to provide names for both metric system and iosystem, I think we're ok.